core/src/main/java/org/springframework/security/core/userdetails/User.java

```diff
@@ -441,7 +441,7 @@ public UserBuilder authorities(GrantedAuthority... authorities) {
 		 */
 		public UserBuilder authorities(Collection<? extends GrantedAuthority> authorities) {
 			Assert.notNull(authorities, "authorities cannot be null");
-			this.authorities = new ArrayList<>(authorities);
```

Comment:
A NullPointerException is not possible in this case since the authorities list is explicitly initialized at line 332 using new ArrayList<>().
```java
		private List<GrantedAuthority> authorities = new ArrayList<>();
```